src/protocol/tx_validity.md

```diff
@@ -183,3 +183,14 @@ If change outputs are present, they must have:
 ### State Changes
 
 Transaction processing is completed by removing spent UTXOs from the state and adding created UTXOs to the state.
+
+### Coinbase
+
+The coinbase is a mechanism for block creators to convert fees into spendable UTXOs.
+
+In order for a coinbase to be valid:
+
+1. It must issued in the form of a [Mint](../protocol/tx_format/transaction.md#TransactionMint) transaction.
+2. The coinbase must be the first transaction within a block.
```

Comment:
```suggestion
2. The coinbase must be the first transaction within a block, even if there are no other transactions in the block and the fee is zero.
```

Reply:
It is how it works now. We can make it optional, but it will require patching the existing logic, and we need to keep it in mind during work with block. 

If the most often case for us is non-zero coinbase transactions, then making it optional may not be worth it because the more complex validation rules could cause bugs in the code.

Reply:
Alright. Both approaches are viable, so whatever works better for you guys.